//regen -b fail

### WorldEdit Version

 WorldEdit version 7.2.9-SNAPSHOT+6064-0ad3ab1

### Platform Version

git-Paper-155 (MC: 1.18.1)

### Confirmations

- [X] I am using the most recent Minecraft release.
- [X] I am using a version of WorldEdit compatible with my Minecraft version.
- [X] I am using the latest or recommended version of my platform software.
- [X] I am NOT using a hybrid server, e.g. a server that combines Bukkit and Forge. Examples include Arclight, Mohist, and Cardboard.
- [X] I am NOT using a fork of WorldEdit, such as FastAsyncWorldEdit (FAWE) or AsyncWorldEdit (AWE)

### Bug Description

If I use //regen -b an error comes up:

```
[14:03:17] [Server thread/WARN]: Regeneration via adapter failed.
java.lang.IllegalStateException: Regen failed.
	at com.sk89q.worldedit.bukkit.adapter.impl.v1_18_R1.PaperweightAdapter.regenerate(PaperweightAdapter.java:584) ~[worldedit-bukkit-7.2.9-SNAPSHOT-dist.jar:?]
	at com.sk89q.worldedit.bukkit.BukkitWorld.regenerate(BukkitWorld.java:216) ~[worldedit-bukkit-7.2.9-SNAPSHOT-dist.jar:?]
	at com.sk89q.worldedit.command.RegionCommands.regenerate(RegionCommands.java:431) ~[worldedit-bukkit-7.2.9-SNAPSHOT-dist.jar:?]
	at com.sk89q.worldedit.command.RegionCommandsRegistration.cmd$_regen(RegionCommandsRegistration.java:576) ~[worldedit-bukkit-7.2.9-SNAPSHOT-dist.jar:?]
	at org.enginehub.piston.CommandManager.execute(CommandManager.java:157) ~[worldedit-bukkit-7.2.9-SNAPSHOT-dist.jar:?]
	at com.sk89q.worldedit.extension.platform.PlatformCommandManager.handleCommand(PlatformCommandManager.java:497) ~[worldedit-bukkit-7.2.9-SNAPSHOT-dist.jar:?]
	at com.sk89q.worldedit.util.eventbus.MethodHandleEventHandler.dispatch(MethodHandleEventHandler.java:51) ~[worldedit-bukkit-7.2.9-SNAPSHOT-dist.jar:?]
	at com.sk89q.worldedit.util.eventbus.EventHandler.handleEvent(EventHandler.java:73) ~[worldedit-bukkit-7.2.9-SNAPSHOT-dist.jar:?]
	at com.sk89q.worldedit.util.eventbus.EventBus.dispatch(EventBus.java:193) ~[worldedit-bukkit-7.2.9-SNAPSHOT-dist.jar:?]
	at com.sk89q.worldedit.util.eventbus.EventBus.post(EventBus.java:181) ~[worldedit-bukkit-7.2.9-SNAPSHOT-dist.jar:?]
	at com.sk89q.worldedit.bukkit.WorldEditPlugin.onCommand(WorldEditPlugin.java:383) ~[worldedit-bukkit-7.2.9-SNAPSHOT-dist.jar:?]
	at com.sk89q.bukkit.util.DynamicPluginCommand.execute(DynamicPluginCommand.java:55) ~[worldedit-bukkit-7.2.9-SNAPSHOT-dist.jar:?]
	at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:159) ~[paper-api-1.18.1-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.craftbukkit.v1_18_R1.CraftServer.dispatchCommand(CraftServer.java:897) ~[paper-1.18.1.jar:git-Paper-163]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.handleCommand(ServerGamePacketListenerImpl.java:2287) ~[?:?]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.handleChat(ServerGamePacketListenerImpl.java:2098) ~[?:?]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.handleChat(ServerGamePacketListenerImpl.java:2079) ~[?:?]
	at net.minecraft.network.protocol.game.ServerboundChatPacket.handle(ServerboundChatPacket.java:46) ~[?:?]
	at net.minecraft.network.protocol.game.ServerboundChatPacket.a(ServerboundChatPacket.java:6) ~[?:?]
	at net.minecraft.network.protocol.PacketUtils.lambda$ensureRunningOnSameThread$1(PacketUtils.java:56) ~[?:?]
	at net.minecraft.server.TickTask.run(TickTask.java:18) ~[paper-1.18.1.jar:git-Paper-163]
	at net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:149) ~[?:?]
	at net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:23) ~[?:?]
	at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:1413) ~[paper-1.18.1.jar:git-Paper-163]
	at net.minecraft.server.MinecraftServer.c(MinecraftServer.java:189) ~[paper-1.18.1.jar:git-Paper-163]
	at net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:122) ~[?:?]
	at net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:1391) ~[paper-1.18.1.jar:git-Paper-163]
	at net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:1384) ~[paper-1.18.1.jar:git-Paper-163]
	at net.minecraft.util.thread.BlockableEventLoop.managedBlock(BlockableEventLoop.java:132) ~[?:?]
	at net.minecraft.server.MinecraftServer.waitUntilNextTick(MinecraftServer.java:1362) ~[paper-1.18.1.jar:git-Paper-163]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1268) ~[paper-1.18.1.jar:git-Paper-163]
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:317) ~[paper-1.18.1.jar:git-Paper-163]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: java.lang.ArrayIndexOutOfBoundsException: Index -521 out of bounds for length 2
	at net.minecraft.util.SimpleBitStorage.get(SimpleBitStorage.java:107) ~[paper-1.18.1.jar:git-Paper-163]
	at net.minecraft.world.level.chunk.PalettedContainer.get(PalettedContainer.java:191) ~[?:?]
	at net.minecraft.world.level.chunk.PalettedContainer.get(PalettedContainer.java:186) ~[?:?]
	at com.sk89q.worldedit.bukkit.adapter.impl.v1_18_R1.PaperweightAdapter.regenForWorld(PaperweightAdapter.java:743) ~[worldedit-bukkit-7.2.9-SNAPSHOT-dist.jar:?]
	at com.sk89q.worldedit.bukkit.adapter.impl.v1_18_R1.PaperweightAdapter.doRegen(PaperweightAdapter.java:635) ~[worldedit-bukkit-7.2.9-SNAPSHOT-dist.jar:?]
	at com.sk89q.worldedit.bukkit.adapter.impl.v1_18_R1.PaperweightAdapter.regenerate(PaperweightAdapter.java:582) ~[worldedit-bukkit-7.2.9-SNAPSHOT-dist.jar:?]
	... 32 more
```

### Expected Behavior

No error and regenerate the area

### Reproduction Steps

My test environment:

world seed: -4753212758496490345
executing location : 7347 83 -5996

1. //chunk
2. //set 0
3. //regen -b
4. error

### Anything Else?

_No response_

--- a/worldedit-bukkit/adapters/adapter-1.18/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/v1_18_R1/PaperweightAdapter.java
+++ b/worldedit-bukkit/adapters/adapter-1.18/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/v1_18_R1/PaperweightAdapter.java
@@ -738,9 +738,7 @@ public final class PaperweightAdapter implements BukkitImplAdapter {
             }
             extent.setBlock(vec, state.toBaseBlock());
             if (options.shouldRegenBiomes()) {
-                PalettedContainer<Biome> biomeIndex = chunk.getSection(chunk.getSectionIndex(vec.getBlockY()))
-                    .getBiomes();
-                Biome origBiome = biomeIndex.get(vec.getBlockX(), vec.getBlockY(), vec.getBlockZ());
+                Biome origBiome = chunk.getNoiseBiome(vec.getX(), vec.getY(), vec.getZ());
                 BiomeType adaptedBiome = adapt(serverWorld, origBiome);
                 if (adaptedBiome != null) {
                     extent.setBiome(vec, adaptedBiome);

